contracts/Pool.sol

```diff
@@ -391,6 +399,74 @@ contract EthPool is Pool {
     }
 }
 
+/// @notice Funding pool contract.
+//// Automatically sends ERC-20 tokens to a configurable set of receivers.
+///
+/// The contract has 2 types of users: the senders and the receivers.
+///
+/// A sender has some tokens and a set of addresses of receivers, to whom he wants to send funds.
+/// In order to send there are 3 conditions, which must be fulfilled:
+///
+/// 1. There must be tokens on his account in this contract.
+///    They can be added with `topUp` and removed with `withdraw`.
+/// 2. Total amount sent to the receivers on each block must be set to a non-zero value.
+///    This is done with `setAmountPerBlock`.
+/// 3. A set of receivers must be non-empty.
+///    Receivers can be added, removed and updated with `setReceiver`.
+///    Each receiver has a weight, which is used to calculate how the total sent amount is split.
+///
+/// Each of these functions can be called in any order and at any time, they have immediate effects.
+/// When all of these conditions are fulfilled, on each block the configured amount is being sent.
+/// It's extracted from the `withdraw`able balance and transferred to the receivers.
+/// The process continues automatically until the sender's balance is empty.
+///
+/// The receiver has an account, from which he can `collect` tokens sent by the senders.
+/// The available amount is updated every `cycleBlocks` blocks,
+/// so recently sent tokens may not be `collect`able immediately.
+/// `cycleBlocks` is a constant configured when the pool is deployed.
+///
+/// A single address can be used both as a sender and as a receiver.
+/// It will have 2 balances in the contract, one with tokens being sent and one with received,
+/// but with no connection between them and no shared configuration.
+/// In order to send received tokens, they must be first `collect`ed and then `topUp`ped
+/// if they are to be sent through the contract.
+///
+/// The concept of something happening periodically, e.g. every block or every `cycleBlocks` are
+/// only high-level abstractions for the user, Ethereum isn't really capable of scheduling work.
+/// The actual implementation emulates that behavior by calculating the results of the scheduled
+/// events based on how many blocks have been mined and only when a user needs their outcomes.
+///
+/// The contract assumes that all amounts in the system can be stored in signed 128-bit integers.
+/// It's guaranteed to be safe only when working with tokens with supply lower than `2 ^ 127`.
```

Comment:
This comment block is now in three places I believe. We should maybe move it to the top of the file, or have it on one contract and reference it from the others.

Reply:
I agree, this is stupid. `@inheritdoc` would be perfect here, but Natspec doesn't support it for the contracts. AFAIK it doesn't have any means of referencing the other items either.

Fixed, now we're just telling the users to go and somehow find the docs for `Pool` by themselves, should be enough :shrug: